azure/services/subnets/subnets_test.go

```diff
@@ -190,6 +190,18 @@ func TestReconcileSubnets(t *testing.T) {
 			expectedError: "",
 			expect: func(s *mock_subnets.MockSubnetScopeMockRecorder, m *mock_subnets.MockClientMockRecorder) {
 				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.Subnet("my-subnet").AnyTimes().Return(infrav1.SubnetSpec{
+					ID:         "subnet-id",
+					Name:       "my-subnet",
+					Role:       infrav1.SubnetNode,
+					CIDRBlocks: []string{"10.0.0.0/16"},
```

Comment:
should we add a case to test that it doesn't overwrite existing NAT Gateways since that's the regression we're fixing. We could also test that if CIDRBlocks and ID are empty in the existing spec, it will set them with the Azure subnet values.